compiler-plugin/src/main/kotlin/arrow/meta/phases/codegen/ir/IrUtils.kt

```diff
@@ -218,13 +232,44 @@ val IrTypeParametersContainer.allTypeParameters: List<IrTypeParameter>
   else
     typeParameters
 
-fun IrMemberAccessExpression.getTypeSubstitutionMap(container: IrTypeParametersContainer): Map<IrTypeParameter, IrType> =
-  container.allTypeParameters.withIndex().associate {
-    it.value to getTypeArgument(it.index)!!
-  }
+fun IrTypeParametersContainer.getTypeSubstitutionMap(expression: IrMemberAccessExpression): Map<IrTypeParameter, Either<KotlinType?, TypeProjection?>> =
```

Comment:
Here I was also keen on return Map<IrTypeParameter, Either<IrSimpleType, IrTypeArgument>>, depending on how we use it in other places, besides Extensions. 
But I can change it to one or the other, it's not fixed.